### PR TITLE
Summary manager tests

### PR DESCRIPTION
**Pull request by arinwt**
Make SummaryManager more testable and add a few tests around the basic behavior.

### CODE REVIEW

File: packages/runtime/container-runtime/src/summaryManager.ts [hunk 1]
```diff
@@ -48,7 +43,60 @@ type ShouldSummarizeState =
...
+                type: summarizerClientType,
+            },
+            [DriverHeader.summarizingClient]: true,
+            [LoaderHeader.reconnect]: false,
+            [LoaderHeader.sequenceNumber]: sequenceNumber,
+        },
+        url: "/_summarizer",
+    };
+
+    const response = await loader.request(request);
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/summaryManager.ts:62]: Any chance you can use requestFluidObject() instead? fluid/component is gone from repo. And ideally we use unified reporting in this space (that logs as much as possible from privacy POV - there are some active bugs in this area for that flow).

File: packages/runtime/container-runtime/src/test/testQuorum.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*!
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/test/testQuorum.ts:1]: This file seems misplaced - shouldn't it be next to quorum implementation?

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -935,15 +935,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
             this.logger,
-            maxOpsSinceLastSummary,
-            this.runtimeOptions.summaryOptions.initialSummarizerDelayMs);
-
-        if (this.connected) {
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            this.summaryManager.setConnected(this.context.clientId!);
-        }
+            requestSummarizer,
+            {
```
Comment (anthony-murphy) [inline packages/runtime/container-runtime/src/containerRuntime.ts:940]: i'd prefer to avoid this ugly conversion and just pass all summaryOptions here

Comment (anthony-murphy) [reply packages/runtime/container-runtime/src/containerRuntime.ts:940]: i also don't like the rename, it makes it really hard to follow the config flow

**Comment by arinwt** [comment]
Closing in favor of #6916 and related PRs #6907, #6908, #6909.
